### PR TITLE
wrapper.sh: wait for child after signaling in exit handler

### LINKED ISSUE #26953: Kind jobs should not immediately kill the cluster on termination before dumping logs
> the info is there, but the failing jobs doesn't have it because the job times out cleaning the kind cluster and the related logs

that is itself a bug, and shouldn't be happening. we should get and handle a TERM from prow well before we get killed and the TERM handler should be capturing logs.

_Originally posted by @BenTheElder in https://github.com/kubernetes/kubernetes/issues/111086#issuecomment-1195791252_

### PR DESCRIPTION
**Pull request by BenTheElder**
should fix #26953

### CODE REVIEW

**Comment by BenTheElder** [comment]
we should consider backporting this to bootstrap / kubekins-e2e image, but first I want to make sure this works well in kind jobs.

File: images/krte/wrapper.sh [hunk 1]
```diff
@@ -47,11 +47,15 @@ cleanup(){
 
 early_exit_handler() {
   >&2 echo "wrapper.sh] [EARLY EXIT] Interrupted, entering handler ..."
-  if [ -n "${WRAPPED_COMMAND_PID:-}" ]; then
+  # if we got here before the wrapped command exited, skip, otherwise signal and wait
+  if [ -n "${WRAPPED_COMMAND_PID:-}" && -z "${EXIT_VALUE:-}" ]; then
```
Comment (aojea) [inline images/krte/wrapper.sh:51]:
```suggestion
  if [ -n "${WRAPPED_COMMAND_PID:-}" ] && [ -z "${EXIT_VALUE:-}" ]; then
```

https://tldp.org/LDP/abs/html/ops.html
> Example 8-3. Compound Condition Tests Using && and ||

Comment (aojea) [reply images/krte/wrapper.sh:51]:
```sh
$ if [ -n "${WRAPPED_COMMAND_PID:-}" && -z "${EXIT_VALUE:-}" ]; then echo "si" ; fi
-bash: [: missing `]'
$ if [ -n "${WRAPPED_COMMAND_PID:-}" ] && [ -z "${EXIT_VALUE:-}" ]; then echo "si" ; fi
si
```

**Pull request merged**